homeassistant/components/squeezebox/coordinator.py

```diff
@@ -70,3 +74,45 @@ def _prepare_status_data(self, data: dict) -> dict:
 
         _LOGGER.debug("Processed serverstatus %s=%s", self.lms.name, data)
         return data
+
+
+class SqueezeBoxPlayerUpdateCoordinator(DataUpdateCoordinator):
+    """Coordinator for Squeezebox players."""
+
+    def __init__(self, hass: HomeAssistant, player: Player) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=player.name,
+            update_interval=timedelta(seconds=PLAYER_UPDATE_INTERVAL),
+            always_update=True,
+        )
+        self.player = player
+        self.available = True
+        self._remove_dispatcher: Callable | None = None
+
+    async def _async_update_data(self) -> dict:
+        """Update the Player() object."""
+        if self.available:
+            # Only update players available at last update, unavailable players are rediscovered instead
+            await self.player.async_update()
+
+            if self.player.connected is False:
+                _LOGGER.debug("Player %s is not available", self.name)
+                self.available = False
+
+                # start listening for restored players
+                self._remove_dispatcher = async_dispatcher_connect(
+                    self.hass, SIGNAL_PLAYER_REDISCOVERED, self.rediscovered
+                )
+        return {}
+
```

Comment:
So how does rediscovery work?

Reply:
We periodically fetch a list of active players from the server. For each player, we check if it is known. If not, we create a `SqueezeBoxEntity` for it. If it is, we send this signal.

In `_async_update_data` in the coordinator, we don't try to update the entity if it was unavailable last time around. This is because a Squeezebox player that is not connected to the server rapidly becomes unknown to the server and the server does not handle inquiries about unknown players gracefully. However, when we mark it disconnected, we start listening for this signal. Thus if it connects again and reappears in the list of players known to the server, we can mark it connected and start polling it again.